Content.Shared/Clothing/EntitySystems/SharedSecurityHailerSystem.cs

```diff
@@ -0,0 +1,411 @@
+    {
+        //Cooldown to prevent spamming
+        //Probably should put a cooldown effect on the mask to show the player, but no idea how to do that !
+        if (_gameTiming.CurTime < ent.Comp.TimeVerbReady)
+            return;
+
+        if (!args.CanAccess || !args.CanInteract || ent.Comp.User != args.User)
+            return;
+
+        //If ERT, they don't switch aggression level
+        if (ent.Comp.IsERT)
+            return;
+
+        var user = args.User;
+
+        args.Verbs.Add(new AlternativeVerb()
+        {
+            Text = Loc.GetString("sec-gas-mask-verb"),
+            Icon = new SpriteSpecifier.Texture(new("/Textures/Interface/VerbIcons/settings.svg.192dpi.png")),
+            Act = () =>
+            {
+                UseVerbSwitchAggression(ent, user);
+            }
+        });
+    }
+
+    private void UseVerbSwitchAggression(Entity<SecurityHailerComponent> ent, EntityUid userActed)
+    {
+        ent.Comp.TimeVerbReady = _gameTiming.CurTime + ent.Comp.VerbCooldown;
+
+        if (!_access.IsAllowed(userActed, ent.Owner))
+        {
+            _sharedAudio.PlayPvs(ent.Comp.SettingError, ent.Owner, AudioParams.Default.WithVariation(0.15f));
+            _popup.PopupEntity(Loc.GetString("sec-gas-mask-wrong_access"), userActed);
+            return;
+        }
+
+        _sharedAudio.PlayPvs(ent.Comp.SettingBeep, ent.Owner, AudioParams.Default.WithVolume(0.5f).WithVariation(0.15f));
+        IncreaseAggressionLevel(ent);
+        Dirty(ent);
+    }
+
+    /// <summary>
+    /// Play the compliance voice line  of the hailer
+    /// </summary>
+    /// <param name="ent"></param>
+    /// <returns>Index of the chosen line from the SoundCollection</returns>
+    protected int PlayVoiceLineSound(Entity<SecurityHailerComponent> ent)
+    {
+        //Move to shared for predictions purposes. Is this good ?
+        var (uid, comp) = ent;
+
+        SoundSpecifier currentSpecifier;
+        if (comp.IsERT)
+            currentSpecifier = comp.ERTAggressionSounds;
+        else if (HasComp<EmaggedComponent>(ent))
+            currentSpecifier = ent.Comp.EmagAggressionSounds;
+        else
+        {
+            currentSpecifier = comp.AggresionLevel switch
+            {
+                AggresionState.Medium => comp.MediumAggressionSounds,
+                AggresionState.High => comp.HighAggressionSounds,
+                _ => comp.LowAggressionSounds,
+            };
+        }
+
+        var resolver = _sharedAudio.ResolveSound(currentSpecifier);
+        if (resolver is not ResolvedCollectionSpecifier collectionResolver)
+            return -1;
+
+        //Replace voice line
+        if (comp.IsHOS && comp.AggresionLevel == AggresionState.High && collectionResolver.Index == comp.SecHailHighIndexForHOS) // MAGIC NUMBER !!!
+        {
+            //There is only one sound replacement at the moment, no need to check indexes
```

Comment:
This logic is really hacky having a bunch of hidden flags. The IsErt + IsHos in particular should really be its own generic data and not hardcoded. The bools also don't even mean anything to a coder because you have to go read the system to know what they are even doing.